### PR TITLE
lowercase pip package names

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This should hopefully address #264

### CODE REVIEW

**Comment by maresb** [comment]
`environment.yaml` is missing

File: tests/test_conda_lock.py [hunk 1]
```diff
@@ -184,6 +184,12 @@ def channel_inversion(tmp_path: Path):
     )
 
 
+@pytest.fixture
+def env_with_uppercase_pip(tmp_path: Path):
+    """Path to an environment.yaml that has a hardcoded channel in one of the dependencies"""
+    return clone_test_dir("test-uppercase-pip", tmp_path).joinpath("environment.yaml")
```
Comment (maresb) [inline tests/test_conda_lock.py:190]:
```suggestion
    return clone_test_dir("test-uppercase-pip", tmp_path).joinpath("environment.yml")
```

File: conda_lock/lookup.py [hunk 2]
```diff
@@ -3,27 +3,42 @@
...
 
     def resolve(self) -> None:
         res = requests.get(self._mapping_url)
         res.raise_for_status()
         self._pypi_lookup = yaml.safe_load(res.content)
+        # lowercase the pypi names
+        assert self._pypi_lookup is not None
+        self._pypi_lookup = {k.lower(): v for k, v in self._pypi_lookup.items()}
+        for v in self._pypi_lookup.values():
+            v["pypi_name"] = v["pypi_name"].lower()
```
Comment (bollwyvl) [inline conda_lock/lookup.py:30]: perhaps also normalize `_` to `-`?

**Comment by maresb** [comment]
It should work with py311, no? I think Mamba was migrated yesterday.

**Comment by mariusvniekerk** [comment]
> It should work with py311, no? I think Mamba was migrated yesterday.

Given the conda-forge a bit more time with the great rebuild before i move the ci to also try 3.11.  It almost certainly works, just not sure if all the dependencies we need exist there yet

File: conda_lock/lookup.py (hunk 2 above)
Comment (mariusvniekerk) [reply conda_lock/lookup.py:30]: yeah that feels prudent

Comment (bollwyvl) [reply conda_lock/lookup.py:30]: Here's "the law": https://peps.python.org/pep-0503/#normalized-names

```
import re

def normalize(name):
    return re.sub(r"[-_.]+", "-", name).lower()
```

**Pull request merged**